### PR TITLE
configurable authz pg connections

### PR DESCRIPTION
**Pull request by bcmdarroch**
### :nut_and_bolt: Description: What code changed, and why?

Aiming to help fix a customer issue where compliance/infra fall over when `ListRulesForProject` times out. 

### :chains: Related Resources

implements a MaxConnections config option for authz [similar to how it's done in authn](https://github.com/chef/automate/blob/master/components/authn-service/tokens/pg/pg.go#L24)

### :+1: Definition of Done
- can set the max number of pg connections authz-service allows

### :athletic_shoe: How to Build and Test the Change

save this as `test.toml`
```
[auth_z]
  [auth_z.v1]
    [auth_z.v1.sys]
      [auth_z.v1.sys.storage]
        max_connections = 5
```

this command should succeed:
```
chef-automate config patch test.toml
```

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/authz-service/storage/postgres/postgres.go [hunk 1]
```diff
@@ -37,12 +37,16 @@ func New(ctx context.Context, migConf migration.Config, dataMigConf datamigratio
...
-func initPostgresDB(ctx context.Context, pgURL string) (*sql.DB, error) {
+func initPostgresDB(ctx context.Context, pgURL string, maxConnections int32) (*sql.DB, error) {
 	d, err := db.PGOpenContext(ctx, pgURL)
 	if err != nil {
 		return nil, err
 	}
 
+	if maxConnections > 0 {
+		d.SetMaxOpenConns(maxConnections)
+	}
```
Comment (stevendanna) [inline components/authz-service/storage/postgres/postgres.go:48]: I'm wonder if we might get a bit more predictable performance by also bumping SetMaxIdleConns.

File: components/authz-service/storage/postgres/postgres.go [hunk 2]
```diff
@@ -37,12 +37,16 @@ func New(ctx context.Context, migConf migration.Config, dataMigConf datamigratio
...
 
-func initPostgresDB(ctx context.Context, pgURL string) (*sql.DB, error) {
+func initPostgresDB(ctx context.Context, pgURL string, maxConnections int32) (*sql.DB, error) {
 	d, err := db.PGOpenContext(ctx, pgURL)
 	if err != nil {
 		return nil, err
 	}
 
+	if maxConnections > 0 {
+		d.SetMaxOpenConns(maxConnections)
```
Comment (stevendanna) [inline components/authz-service/storage/postgres/postgres.go:47]: I think you'll need to convert this to an int here or at some point in the call chain

```suggestion
		d.SetMaxOpenConns(int(maxConnections))
```

File: components/authz-service/storage/postgres/postgres.go (hunk 1 above)
Comment (jaym) [reply components/authz-service/storage/postgres/postgres.go:48]: what do you think about having them be the same

Comment (stevendanna) [reply components/authz-service/storage/postgres/postgres.go:48]: For a small number of connections, I would advocate having them be the same.

File: components/authz-service/storage/postgres/postgres.go (hunk 2 above)
Comment (bcmdarroch) [reply components/authz-service/storage/postgres/postgres.go:47]: we're setting it to `int` here: https://github.com/chef/automate/pull/3442/files#diff-7b11e67859dabc4eb026fda620cc2815R23

**Pull request merged**